app/controllers/areas_controller.rb

```diff
@@ -18,7 +18,11 @@ def index
   end
 
   def search
-    api_response = Imminence.mapit_api.location_for_postcode(params[:postcode])
+    # Strip trailing whitespace, most non-alphanumerics, and use the
+    # uk_postcode gem to potentially transpose O/0 and I/1.
+    sanitized_postcode = UKPostcode.parse(params[:postcode].gsub(/[^\w\s]/i, '').strip).to_s
```

Comment:
I've made it into a `PostcodeSanitizer` class with a `sanitize` method that takes the postcode, and written tests for each of the different things it does. Thanks!